ctk/common/src/test/kotlin/org/codice/compilance/verification/core/SamlDefinedIdentifiersVerifierSpec.kt

```diff
@@ -0,0 +1,281 @@
+            createResponse(
+                    identifierValue = "&lt;example.email@domain.com&gt;",
+                    identifierFormat = NAME_ID_FORMAT_EMAIL).let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_3_2_a.message)
+            }
+        }
+
+        /* 8.3.6 Entity Identifier */
+        "valid Entity name identifier" {
+            createResponse(
+                    identifierValue = validEntityId,
+                    identifierFormat = ENTITY).let {
+                SamlDefinedIdentifiersVerifier(it).verify()
+            }
+        }
+
+        "invalid NameQualifier attribute on Entity name identifier" {
+            createResponse(
+                    identifierValue = validEntityId,
+                    identifierFormat = ENTITY,
+                    extraIdentifierAttribute = """NameQualifier="$validEntityId"""").let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_3_6_a.message)
+            }
+        }
+
+        "invalid SPNameQualifier attribute on Entity name identifier" {
+            createResponse(
+                    identifierValue = validEntityId,
+                    identifierFormat = ENTITY,
+                    extraIdentifierAttribute = """SPNameQualifier="$validEntityId"""").let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_3_6_a.message)
+            }
+        }
+
+        "invalid SPProvidedID attribute on Entity name identifier" {
+            createResponse(
+                    identifierValue = validEntityId,
+                    identifierFormat = ENTITY,
+                    extraIdentifierAttribute = """SPProvidedID="$validEntityId"""").let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_3_6_a.message)
+            }
+        }
+
+        "valid length Entity name identifier" {
+            createResponse(
+                    identifierValue = maxLengthEntityId,
+                    identifierFormat = ENTITY).let {
+                SamlDefinedIdentifiersVerifier(it).verify()
+            }
+        }
+
+        "invalid length Entity name identifier" {
+            createResponse(
+                    identifierValue = maxLengthEntityId + "A",
+                    identifierFormat = ENTITY).let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_3_6_b.message)
+            }
+        }
+    }
+}
```

Comment:
❗️ `verifyPersistentIdentifiers` and `verifyTransientIdentifiers` are not tested at all.

Reply:
Those verifications were merged in after my PR was up.